sources/editor/Xenko.Assets.Presentation/AssetEditors/ScriptEditor/SimpleCodeTextEditor.cs

```diff
@@ -185,8 +185,12 @@ private async void CaretOnPositionChanged(object sender, EventArgs eventArgs)
             var caretOffset = CaretOffset;
             if (caretOffset <= text.Length)
             {
-                var result = await braceMatchingService.GetAllMatchingBracesAsync(document, caretOffset, token).ConfigureAwait(true);
-                braceMatcherHighlighter.SetHighlight(result.leftOfPosition, result.rightOfPosition);
+                try
```

Comment:
In your experience, can the token be already cancelled at that point?

If so, checking `if (token.IsCancellationRequested)` might save some computation (no await state-machine, no exception). Of course, the `try...catch` is still necessary.

Reply:
Looking at the whole code more closely, it could happen in a large document since there is an async call to `document.GetTextAsync()` a few lines above.

So while we are at it, let's give the `token` to the `document.GetTextAsync()` call as well and put a `try...catch` around all of it. In which case, testing the token is not necessary anymore.